include/xtensor/xcontainer.hpp

```diff
@@ -101,19 +90,6 @@ namespace xt
         const inner_strides_type& strides() const noexcept;
         const inner_backstrides_type& backstrides() const noexcept;
 
-        void transpose();
-
-        template <class S, class Tag = check_policy::none>
-        void transpose(S&& permutation, Tag check_policy = Tag());
-
-#ifdef X_OLD_CLANG
-        template <class I, class Tag = check_policy::none>
-        void transpose(std::initializer_list<I> permutation, Tag check_policy = Tag());
-#else
-        template <class I, std::size_t N, class Tag = check_policy::none>
-        void transpose(const I (&permutation)[N], Tag check_policy = Tag());
-#endif
-
```

Comment:
I wonder if we should keep the `transpose` methods for containers with dynamic layout. That would allow "in-place" transposition, but may be confusing with the methods provided in `xstridedview.hpp`.